tests/unit/core/reducers/test_addons.js

```diff
@@ -553,6 +558,289 @@ describe(__filename, () => {
+      const state = addons(initialState, _updateRatingCounts());
+
+      expect(state).toEqual(initialState);
+    });
+
+    it('increments only the rating count for a newly added score', () => {
+      const ratingCount = 300;
+      const reviewCount = 200;
+      const addon = addonWithRatings({
+        count: ratingCount,
+        text_count: reviewCount,
+      });
+
+      let state = initStateWithAddon(addon);
+      state = addons(
+        state,
+        _updateRatingCounts({
+          addonId: addon.id,
+          oldReview: null,
+          newReview: createInternalReview({
+            ...fakeReview,
+            body: null,
+            score: 5,
+          }),
+        }),
+      );
+
+      const storedAddon = state.byID[addon.id];
+      expect(storedAddon.ratings.count).toEqual(ratingCount + 1);
+      // Since a review without a body was added, no change.
+      expect(storedAddon.ratings.text_count).toEqual(reviewCount);
+    });
+
+    it('increments the review count for newly added bodies', () => {
+      const reviewCount = 200;
+      const addon = addonWithRatings({
+        text_count: reviewCount,
+      });
+
+      let state = initStateWithAddon(addon);
+      state = addons(
+        state,
+        _updateRatingCounts({
+          addonId: addon.id,
+          oldReview: null,
+          newReview: createInternalReview({
+            ...fakeReview,
+            body: 'Fantastic add-on',
+            score: 5,
+          }),
+        }),
+      );
+
+      expect(state.byID[addon.id].ratings.text_count).toEqual(reviewCount + 1);
+    });
+
+    it('increments the review count for reviews updated to include a body', () => {
+      const reviewCount = 200;
+      const addon = addonWithRatings({
+        text_count: reviewCount,
+      });
+
+      let state = initStateWithAddon(addon);
+      state = addons(
+        state,
+        _updateRatingCounts({
+          addonId: addon.id,
+          oldReview: createInternalReview({
+            ...fakeReview,
+            body: null,
+            score: 5,
+          }),
+          newReview: createInternalReview({
+            ...fakeReview,
+            body: 'Fantastic add-on',
+            score: 5,
+          }),
+        }),
+      );
+
+      expect(state.byID[addon.id].ratings.text_count).toEqual(reviewCount + 1);
+    });
+
+    it('does not increment rating or review counts for existing reviews', () => {
+      const ratingCount = 100;
+      const reviewCount = 200;
+      const addon = addonWithRatings({
+        count: ratingCount,
+        text_count: reviewCount,
+      });
+
+      let state = initStateWithAddon(addon);
+      state = addons(
+        state,
+        _updateRatingCounts({
+          addonId: addon.id,
+          oldReview: createInternalReview({ ...fakeReview }),
+          newReview: createInternalReview({ ...fakeReview }),
+        }),
+      );
+
+      expect(state.byID[addon.id].ratings.text_count).toEqual(reviewCount);
```

Comment:
According to the test case description, there is a missing assertion for `ratingCount`, no?

Reply:
Good catch, thanks